src/components/course/data/utils.jsx

```diff
@@ -149,74 +149,102 @@ export function findCouponCodeForCourse(couponCodes, catalogList = []) {
   }));
 }
 
+/**
+ * Determines the sort order of two enterprise offers based on the
+ * specified property for the user. Returns -1 if the first offer should
+ * be prioritized, 1 if the second offer should be prioritized, and 0 if
+ * there is no difference in priority.
+ *
+ * @param {object} args
+ * @param {object} args.firstOffer An enterprise offer to compare with the second offer.
+ * @param {object} args.secondOffer An enterprise offer to compare with the first offer.
+ *
+ * @returns The sort comparison value.
+ */
+export const compareOffersByProperty = ({ firstOffer, secondOffer, property }) => {
+  const firstOfferValue = firstOffer[property];
+  const secondOfferValue = secondOffer[property];
+  if (firstOfferValue && secondOfferValue) {
+    if (firstOfferValue < secondOfferValue) {
+      return -1;
+    }
+    if (firstOfferValue >= secondOfferValue) {
+      return 1;
+    }
+  }
+  if (!firstOfferValue && secondOfferValue) {
+    return -1;
+  }
+  if (firstOfferValue && !secondOfferValue) {
+    return 1;
+  }
+  return 0;
```

Comment:
If we're only concerned with getting positive/negative/0 values back (which [sort](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/sort) is), can be a one-liner.
```suggestion
  return (firstOffer[property] || 0) - (secondOffer[property] || 0);
```

Reply:
@marlonkeating I'm not sure that suggestion is logically the same; it fails the tests:

<img width="763" alt="image" src="https://github.com/openedx/frontend-app-learner-portal-enterprise/assets/2828721/544d05b2-7262-4962-875b-58e49a94a6f7">

Reply:
It's not exactly the same, you'd need to update unit tests that expect the compare function to return 1/-1.  

But considering that this compare function is at least currently only used for javascript array's sort function (which only cares if the compare function returns a positive or negative value) the one-liner should work.